pkg/instancetype/instancetype_test.go

```diff
@@ -1442,4 +1444,240 @@ var _ = Describe("Instancetype and Preferences", func() {
 			})
 		})
 	})
+
+	Context("preference requirements check", func() {
+		DescribeTable("should pass when sufficient resources are provided", func(instancetypeSpec *instancetypev1alpha2.VirtualMachineInstancetypeSpec, preferenceSpec *instancetypev1alpha2.VirtualMachinePreferenceSpec, vmiSpec *v1.VirtualMachineInstanceSpec) {
+			path, err := instancetypeMethods.CheckPreferenceRequirements(instancetypeSpec, preferenceSpec, vmiSpec)
+			Expect(path).To(BeNil())
+			Expect(err).ToNot(HaveOccurred())
```

Comment:
I think you should add another check for type of error (because it can return error for CPU or memory)

Reply:
`Expect(err).ToNot(HaveOccurred())`  - I'm not sure what else I can check here?

Reply:
Sorry, I mismatch ToNot with To :D